patches/0004-Add-OpenSSL-crypto-backend.patch

```diff
@@ -518,6 +518,41 @@ index 00000000000000..69af0ffe2fcf80
 +func VerifyEd25519(pub *PublicKeyEd25519, message, sig []byte) error {
 +	return openssl.VerifyEd25519(pub, message, sig)
 +}
++
++func supportsRSAPrimes(primes int) bool {
++	// The built-in OpenSSL 3 providers and OpenSSL 1 do support n-prime RSA keys,
++	// but the SymCrypt provider for OpenSSL only supports 2-prime RSA keys.
++	// Only 2-prime RSA keys are FIPS compliant, other n having compatibility
++	// and security issues. Even crypto/rsa deprecated rsa.GenerateMultiPrimeKey as of Go 1.21.
++	// Given the above reasons, we only support 2-prime RSA keys.
++	return primes == 2
++}
++
++func supportsPKCS1v15Hash(h crypto.Hash) bool {
++	if h == 0 {
++		// The SymCrypt provider for OpenSSL does not support unpadded PKCS1v15 signatures,
++		// although the built-in OpenSSL providers do.
++		// Given that unpadded PKCS1v15 signatures are not widely used, do not support them.
++		return false
++	}
++	return supportsRSAHash(h)
++}
++
++func supportsRSAHash(h crypto.Hash) bool {
++	if !SupportsHash(h) {
++		return false
++	}
++	// The SymCrypt provider for OpenSSL only supports digest function provided by SymCrypt itself.
++	// It can happen that SupportsHash return true for a digest not provided by SymCrypt, but by another provider.
++	// The RSA operations will fail if that happens.
++	// To avoid this case we only support digest functions also supported by SymCrypt, which happens to be
++	// the most common ones.
++	return h == crypto.SHA1 || h == crypto.SHA256 || h == crypto.SHA384 || h == crypto.SHA512
```

Comment:
This seems like it could be significant for someone using non-SymCrypt. (Ubuntu Pro?) FIPS guide update?

(Also, even though we don't have automation for https://github.com/microsoft/go/tree/microsoft/main/docs yet, we could start a doc for 1.24.)

Reply:
I'll revert this particular change. The SymCrypt team agreed on fixing it on their side.